x/auth/vesting/cmd/vestcalc/vestcalc.go

```diff
@@ -0,0 +1,532 @@
+		r = r + time.Second
+		frac := r.String()
+		// append skipping the leading "1"
+		return s + frac[1:] // adds the suffix
+	}
+	return s + "s" // now the suffix
+}
+
+// maxTime gives the maximum of a set of times, or the zero time if empty.
+func maxTime(cliffs []time.Time) time.Time {
+	tm := time.Time{}
+	for _, c := range cliffs {
+		if c.After(tm) {
+			tm = c
+		}
+	}
+	return tm
+}
+
+var (
+	validDenoms = map[string]bool{"ubld": true}
+)
+
+// Custom flag types
+
+// isoDate is time.Time as a flag.Value in shortIsoFmt.
+type isoDate struct{ time.Time }
+
+// Set implements flag.Value.Set().
+func (id *isoDate) Set(s string) error {
+	t, err := time.ParseInLocation(shortIsoFmt, s, time.Local)
+	if err != nil {
+		return err
+	}
+	id.Time = t
+	return nil
+}
+
+// String implements flag.Value.String().
+func (id *isoDate) String() string {
+	return id.Format(shortIsoFmt)
+}
+
+// isoDateFlag makes a new isoDate flag, accessed as a time.Time.
+func isoDateFlag(name string, usage string) *time.Time {
+	id := isoDate{time.Time{}}
+	flag.CommandLine.Var(&id, name, usage)
+	return &id.Time
+}
+
+// isoDateList is []time.Time as a flag.Value in repeated or comma-separated shortIsoFmt.
+type isoDateList []time.Time
+
+// Set implements flag.Value.Set().
+func (dates *isoDateList) Set(s string) error {
+	for _, ds := range strings.Split(s, ",") {
+		d, err := time.ParseInLocation(shortIsoFmt, ds, time.Local)
+		if err != nil {
+			return err
+		}
+		*dates = append(*dates, d) // accumulates repeated flag arguments
+	}
+	return nil
+}
+
+// String implements flag.Value.String().
+func (dates *isoDateList) String() string {
+	s := ""
+	for _, t := range *dates {
+		if s == "" {
+			s = t.Format(shortIsoFmt)
+		} else {
+			s = s + "," + t.Format(shortIsoFmt)
+		}
+	}
+	return s
+}
+
+// isoDateListFlag makes a new isoDateList flag.
+func isoDateListFlag(name string, usage string) *isoDateList {
+	dates := isoDateList([]time.Time{})
+	flag.Var(&dates, name, usage)
+	return &dates
+}
+
+// isoTime is time.Time as a flagValue in HH:MM format.
+type isoTime struct{ time.Time }
+
+// Set implements flag.Value.Set().
+func (it *isoTime) Set(s string) error {
+	t, err := time.Parse(hhmmFmt, s)
+	if err != nil {
+		return err
+	}
+	it.Time = t
+	return nil
+}
+
+// String implements flag.Value.String().
+func (it *isoTime) String() string {
+	return it.Format(hhmmFmt)
+}
+
+// isoTimeFlag makes a new isoTime flag, accessed as a time.Time.
+func isoTimeFlag(name string, value string, usage string) *time.Time {
+	t, err := time.Parse(hhmmFmt, value)
+	if err != nil {
+		t = time.Time{}
+	}
+	it := isoTime{t}
+	flag.CommandLine.Var(&it, name, usage)
+	return &it.Time
+}
+
+var (
+	flagStart  = isoDateFlag("start", "Start date for the vesting in format 2006-01-02T15:04 (local time).")
+	flagMonths = flag.Int("months", 1, "Number of months to vest over.")
+	flagAmount = flag.Int64("amount", 0, "Total amount to vest.")
```

Comment:
In fact, a better design might be to accept `sdk.Coins` as the amount rather than split amount and denom (i.e. `"100ubld"` would be a valid amount, as well as `"100ubld,30urun"`).  I'm not sure what the vesting code is able to handle, though.

Reply:
Ah, it looks like vesting accounts handle an arbitrary collection of coins - I'll update to handle the more general case of multiple amounts.